File: querydsl-sample/java-module/src/test/java/io/philo/QueryIntegrationTest.java
package io.philo;

import io.philo.app.JEntity;
import io.philo.app.JEntityDto;
import io.philo.app.JEntityQuery;
import io.philo.app.JEntityRepository;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;

import java.util.List;

import static org.junit.jupiter.api.Assertions.assertAll;
import static org.junit.jupiter.api.Assertions.assertEquals;

@SpringBootTest
public class QueryIntegrationTest {

    @Autowired
    JEntityQuery query;

    @Autowired
    JEntityRepository repository;

    @Test
    void query() {

        repository.save(new JEntity("some entity"));

        List<JEntityDto> dtos = query.searchAll();

        assertAll(
                () -> assertEquals(dtos.size(), 1),
                () -> assertEquals(dtos.get(0).getDescription(), "some entity")
        );
    }
}

File: querydsl-sample/kotlin-module/src/test/kotlin/io/philo/QueryIntegrationTest.kt
package io.philo

import io.kotest.matchers.shouldBe
import io.philo.app.KEntity
import io.philo.app.KEntityQuery
import io.philo.app.KEntityRepository
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest

@SpringBootTest
class QueryIntegrationTest {

    @Autowired
    lateinit var query: KEntityQuery

    @Autowired
    lateinit var repository: KEntityRepository

    @Test
    fun query() {

        repository.save(KEntity("some entity"))

        val dtos = query.searchAll()

        dtos.size shouldBe 1
        dtos[0].description shouldBe "some entity"
    }
}